### PR TITLE
Add platform host-based unit tests and code coverage

### PR DESCRIPTION
**Pull request by Javagedes**
## Description

Adds a `PlatformTest.py` per platform package which:

1. self verifies the platform host-based unit test dsc is up to date (no new hosted-based test INFs have been added locally or from a submodule that test source used by the platform).
2. Compiles the host-based unit tests.
3. Executes the host-based unit tests.
4. Optionally generates code coverage data.

Adds additional pipeline jobs to run the host-based tests and generate code coverage information.

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

CI

## Integration Instructions

CI

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

File: .azurepipelines/Platform-Build-GCC5.yml [hunk 1]
```diff
@@ -176,3 +177,14 @@ jobs:
...
+
+      QemuSbsa_CODE_COVERAGE:
+        BuildPackage: QemuSbsaPkg
+        BuildFile: "Platforms/QemuSbsaPkg/Test/PlatformTest.py"
+        BuildArch: "AARCH64"
+        BuildFlags: "CODE_COVERAGE=TRUE CC_FULL=TRUE CC_FLATTEN=TRUE"
+        BuildTarget: "NOOPT"
+        BuildExtraTag: "UNIT_TESTS"
+        BuildExtraStep:
+          - script: echo no steps provided
```
Comment (makubacki) [inline .azurepipelines/Platform-Build-GCC5.yml:189]: I think this is a bit more clear:

```suggestion
          - script: echo no extra steps provided

```

File: Platforms/QemuQ35Pkg/Test/PlatformTest.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
+from collections import namedtuple
```
Comment (makubacki) [inline Platforms/QemuQ35Pkg/Test/PlatformTest.py:1]: Normally we add a copyright header to Python files.

**Review by makubacki — APPROVED** [review]

File: .azurepipelines/Platform-Build-GCC5.yml (hunk 1 above)
Comment (Javagedes) [reply .azurepipelines/Platform-Build-GCC5.yml]: Updated for clarity. Thanks!

File: Platforms/QemuQ35Pkg/Test/PlatformTest.py (hunk 2 above)
Comment (Javagedes) [reply Platforms/QemuQ35Pkg/Test/PlatformTest.py:7]: Good catch. I've added this to both files. Thanks!

**Pull request merged**